src/System.Data.SqlClient/tests/ManualTests/SQL/AsyncTest/ReadAsyncTest.cs

```diff
@@ -0,0 +1,107 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Data.SqlClient.ManualTesting.Tests
+{
+    public class ReadAsyncTest
+    {
+        [CheckConnStrSetupFact]
+        public static void Run()
+        {
+            CompareSyncAndAsync();
+        }
+
+        public static async void CompareSyncAndAsync()
+        {
+            double elapsedAsync = await RunReadAsync();
+            double elapsedSync = RunRead();
+            Assert.True(elapsedAsync < (elapsedSync / 2.0d));
```

Comment:
It took about 15~30ms in my environment for asynchronous operation.
For synchronous read, it took about 2000ms.

Reply:
I checked, and `TryProcessDone()` was not retried after my change. However, I have just became curious if I force it to retry `TryProcessDone()` by following code:
```
Task<bool> t1, t2;
do
{
    t1 = reader.ReadAsync();
    t2 = reader.ReadAsync();
}
while (await t1 && await t2);
```

Reply:
> It took about 15~30ms in my environment for asynchronous operation.
For synchronous read, it took about 2000ms.

Based on this can you provide a better Assert for this test. 
My concern is that if ReadSync takes 30 seconds in some cases, then ReadAsync should still take ~30 ms. So a proportionate comparison with the time taken by ReadAsync and  ReadSync doesn't make sense

Reply:
> Task<bool> t1, t2;
do
{
    t1 = reader.ReadAsync();
    t2 = reader.ReadAsync();
}
while (await t1 && await t2);

This Code will not retry Done token processing in the middle of the TDS stream.